fix(Table): fixed empty and shadow issues when virtual mode

linxianxi opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is confirmed.
Your pull requests will be merged after one of the collaborators approve.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/48991
fix https://github.com/ant-design/ant-design/issues/48959
fix https://github.com/ant-design/ant-design/issues/49185

### 💡 Background and solution

<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list the final API implementation and usage sample if that is a new feature.
-->

### 📝 Changelog

<!--
Describe changes from the user side, and list all potential break changes or other risks.
--->

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fixed empty and shadow issues when virtual mode      |
| 🇨🇳 Chinese |     修复虚拟表格下的空数据和阴影问题      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

linxianxi commented:
> 看下用例？

这个怎么解决？我跑 test -u 了

afc163 commented on docs/react/faq.zh-CN.md:
```diff
@@ -107,14 +107,14 @@ antd 内部会对 props 进行浅比较实现性能优化。当状态变更，
-| AntV | https://antv.antgroup.com |
+| Ant Design Mini   | https://ant-design-mini.antgroup.com      |
+| Ant Design Charts | https://ant-design-charts.antgroup.com    |
+| AntV              | https://antv.antgroup.com                 |
```
不应该动这个文件。

linxianxi replied:
rebase 的时候动到了，已删。帮我看看 demo test failed 咋搞。

linxianxi committed:
fix(Table): fixed empty and shadow issues when virtual mode

linxianxi committed:
Merge branch 'feature' of github.com:ant-design/ant-design into fix-table

afc163 commented:
在 React 16 下貌似 snapshot 对不上。https://github.com/ant-design/ant-design/actions/runs/9342827135/job/25711683702?pr=49080

<img width="1218" alt="图片" src="https://github.com/ant-design/ant-design/assets/507615/163e7615-23fa-4200-a99c-ddd7e7e9caa0">

linxianxi commented:
> #48959

没看出来改动跟不同 react 版本有什么影响

linxianxi commented:
> 在 React 16 下貌似 snapshot 对不上。https://github.com/ant-design/ant-design/actions/runs/9342827135/job/25711683702?pr=49080
> 
> <img alt="图片" width="1218" src="https://private-user-images.githubusercontent.com/507615/335920318-163e7615-23fa-4200-a99c-ddd7e7e9caa0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTczODIwMzgsIm5iZiI6MTcxNzM4MTczOCwicGF0aCI6Ii81MDc2MTUvMzM1OTIwMzE4LTE2M2U3NjE1LTIzZmEtNDIwMC1hOTljLWRkZDdlN2U5Y2FhMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QwMjI4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMzU4ZjVjYmRmNjU4ZDg5MGU4OWY3OTIyZjY0ZWViZjkwODM3ZjZmOTIwYzQ1ZmQ0ZGM4OTEzNTM1OGZmOTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.SnIsXApkspGIIGpcxrr5b41kEGg06GoD8dbiqgZ1-l8">

没看出来改动跟不同 react 版本有什么影响

zombieJ committed:
test: update test case

zombieJ committed:
chore: update package.json

zombieJ committed:
test: ignore useless test

zombieJ committed:
chore: fix lint

afc163 commented on components/table/__tests__/demo-extend.test.ts:
```diff
@@ -1,3 +1,13 @@
+    'sticky.tsx',
+    'fixed-gapped-columns.tsx',
+    'fixed-columns.tsx',
+    'fixed-columns-header.tsx',
```
为啥把这些 skip 了？

zombieJ replied:
extend test 测试会注入 `rc-trigger` 强制开窗，Table 里除了过滤其实都用不到。这几个 test case 有滚动条，会遇到 React 16 & 17 与 React 18 在 jsdom 下行为不一致的问题（真实世界则是一致的）。所以直接 skip 就行了。

afc163 replied:
这样不好吧，以后别的问题也看不出来 snapshot diff 了。

zombieJ replied:
snapshot test 有两个，一个是 `demo.test` 一个是 `demo-extend.test`，`demo.test` 还是在的。`demo-extend.test` 是给 `rc-trigger` 测试用的，只是 Trigger 做注入没有考虑其他情况。

afc163 replied:
👌🏻 

我一直不知道 demo-extend 是干啥的。

linxianxi commented:
这个跟上 5.18.0 吧

linxianxi committed:
Merge branch 'feature' of github.com:ant-design/ant-design into fix-table

linxianxi committed:
chore: bump rc-table

afc163 commented:
@SocketSecurity ignore-all

afc163 committed:
Merge branch 'feature' into fix-table

afc163 committed:
Merge branch 'feature' into fix-table

yoyo837 commented on package.json:
```diff
@@ -154,7 +154,7 @@
     "rc-steps": "~6.0.1",
     "rc-switch": "~4.1.0",
-    "rc-table": "~7.45.7",
+    "rc-table-test-image-diff": "^1.2.0",
```
这是整了个啥

linxianxi replied:
自己 fork 的做测试

li-jia-nan commented on package.json:
```diff
@@ -154,7 +154,7 @@
     "rc-steps": "~6.0.1",
     "rc-switch": "~4.1.0",
-    "rc-table": "~7.45.7",
+    "rc-table": "~7.46.1",
```
```suggestion
    "rc-table": "~7.47.1",
```

li-jia-nan committed:
Merge branch 'feature' into fix-table

Signed-off-by: lijianan <574980606@qq.com>

afc163 committed:
fix: Checkbox indeterminate style (#50083)

afc163 committed:
fix: Tour mobile view (#50082)

* fix: Tour mobile view

* Update .husky/pre-commit

Signed-off-by: afc163 <afc163@gmail.com>

---------

Signed-off-by: afc163 <afc163@gmail.com>

dependabot[bot] committed:
chore: bump the dev-dependencies group with 1 update (#50077)

* chore: bump the dev-dependencies group with 1 update

Bumps the dev-dependencies group with 3 updates: [@antv/g6](https://github.com/antvis/g6), [eslint-plugin-lodash](https://github.com/idok/eslint-plugin-lodash) and [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn).


Updates `@antv/g6` from 4.8.24 to 5.0.10
- [Release notes](https://github.com/antvis/g6/releases)
- [Changelog](https://github.com/antvis/G6/blob/v5/CHANGELOG.md)
- [Commits](https://github.com/antvis/g6/commits)

Updates `eslint-plugin-lodash` from 7.4.0 to 8.0.0
- [Changelog](https://github.com/idok/eslint-plugin-lodash/blob/master/CHANGELOG.md)
- [Commits](https://github.com/idok/eslint-plugin-lodash/commits/v8.0.0)

Updates `eslint-plugin-unicorn` from 54.0.0 to 55.0.0
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v54.0.0...v55.0.0)

---
updated-dependencies:
- dependency-name: "@antv/g6"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-lodash
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-unicorn
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update package.json

Signed-off-by: lijianan <574980606@qq.com>

* Update package.json

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: lijianan <574980606@qq.com>

thinkasany committed:
site: improve color type (#50089)

* type(theme): improve color type

* ThemeColorPickerProps

* lint-fix

LLmoskk committed:
fix: should not display overlay when the content is null/undefined (#50064)

* fix: content结果为null的情况不显示overlay

* fix: 兼容undefined与null

* Update index.tsx

Signed-off-by: lijianan <574980606@qq.com>

* Update getRenderPropValue.ts

Signed-off-by: lijianan <574980606@qq.com>

* chore: update snap

* Update index.test.tsx

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: H JY <huangjunyi@ama-auto.com>
Co-authored-by: lijianan <574980606@qq.com>

afc163 committed:
Merge branch 'feature' into fix-table

afc163 committed:
Update package.json

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: afc163 <afc163@gmail.com>

afc163 commented:
ci 挂了

linxianxi commented:
> Details

rc-util 里 findDOMNode 里的 instanceOf 在 image diff 有问题，应该还没解决，主要是这个问题

vagusX committed:
ci: current workflow job url (#50090)

thinkasany committed:
site: use overlayStyle instead of popupStyle (#50095)

thinkasany committed:
chore(deps-dev): bump husky from 9.0.11 to 9.1.2 (#50092)

li-jia-nan committed:
refactor: rewrite Overlay of PurePanel (#50097)

thinkasany committed:
docs: fix typos (#50096)

* docs: fix typos

* fix

guoyunhe committed:
feat(Upload): showUploadList add extra option (#50098)

vagusX committed:
ci: fix workflow url for Visual Regression Diff (#50105)

li-jia-nan committed:
site: update site/demo style (#50107)

* site: update site/demo style

* fix: fix lint

vagusX committed:
ci: we can skip a single file uploading error when uploading a folder… (#50108)

* ci: we can skip a single file uploading error when uploading a folder in scripts/visual-regression/upload.js

* chore: update

ang33l committed:
fix: add missed translation in pl_PL (#50023)

* Update pl_PL.ts

Added most translations for polish language

Signed-off-by: Aleksander Marek <45952400+ang33l@users.noreply.github.com>

* Update pl_PL.ts

Signed-off-by: lijianan <574980606@qq.com>

* Update index.test.tsx.snap

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: Aleksander Marek <45952400+ang33l@users.noreply.github.com>
Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: afc163 <afc163@gmail.com>
Co-authored-by: lijianan <574980606@qq.com>

li-jia-nan committed:
site: rewrite with CSS logical properties (#50109)

* site: use CSS logical properties

* Update components/anchor/demo/targetOffset.tsx

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: lijianan <574980606@qq.com>

* fix: fix

* fix: fix

* chore: fix

* fix: add more

* fix: add more

---------

Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: afc163 <afc163@gmail.com>

li-jia-nan committed:
type: use DirectionType (#50117)

li-jia-nan committed:
type: code optimization (#50116)

li-jia-nan committed:
site: rewrite with CSS logical properties again (#50113)

* site: use CSS logical properties

* Update components/anchor/demo/targetOffset.tsx

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: lijianan <574980606@qq.com>

* fix: fix

* fix: fix

* chore: fix

* fix: add more

* fix: add more

* site: rewrite with CSS logical properties

---------

Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: afc163 <afc163@gmail.com>

zombieJ committed:
feat: ColorPicker support gradient color (#50050)

* refactor: support type update

* chore: update clear style

* chore: bump color picker

* chore: use slider

* chore: bump color picker

* chore: range slider

* chore: layout

* chore: useModeColor

* chore: simplify

* chore: bump color picker

* refactor: event

* chore: tmp lock check

* chore: of it

* chore: update ts def

* chore: update ts def

* chore: remove useless ts

* chore: linear

* chore: adjust style

* chore: rm useless code

* chore: fill color

* chore: basic linear

* chore: support toStr

* chore: limit minCount

* chore: use cache

* chore: drag support:

* chore: yes

* chore: update demo

* chore: useLayoutEffect instead

* chore: fix click to add point

* chore: add smmoth

* chore: support of locale

* chore: add locale

* chore: fix lint

* chore: adjust style

* chore: fix lint

* chore: fix style

* test: fix test case

* chore: fix popover

* test: fix test case

* chore: fix test

* test: clean up

* chore: fix lint

* chore: fix lint

* chore: fix lint

* test: coverage

* test: coverage

* test: coverage

* test: coverage

* test: coverage

* test: coverage

* chore: fix docs

* docs: update demo desc

* chore: enhance hover range

* fix: delete not working

* chore: fix lint

* test: coverage

* test: coverage

* chore: clean up

* chore: adjust

* chore: highlight

* chore: adjust style

* chore: fix lint

* chore: update demo

* chore: memo perf

* refactor: up to down colors

* test: update snapshot

xrkffgg committed:
docs: add changelog 5.19.4 (#50129)

* docs: add changelog 5.19.4

* docs: update

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: xrkffgg <xrkffgg@vip.qq.com>

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: xrkffgg <xrkffgg@vip.qq.com>

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: xrkffgg <xrkffgg@vip.qq.com>

* docs: update

---------

Signed-off-by: xrkffgg <xrkffgg@vip.qq.com>
Co-authored-by: afc163 <afc163@gmail.com>

zombieJ committed:
test: update snapshot (#50136)

xrkffgg committed:
docs: update publish date (#50142)

github-actions[bot] committed:
chore: upgrade deps (#50141)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

zombieJ committed:
fix: popup align position (#50134)

* fix: popup align

* chore: adjust tooltip

* fix: var

zombieJ committed:
ci: skip artifact check (#50150)

dependabot[bot] committed:
chore: bump the dev-dependencies group with 1 update (#50133)

* chore: bump the dev-dependencies group with 1 update

Bumps the dev-dependencies group with 3 updates: [@antv/g6](https://github.com/antvis/g6), [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) and [eslint-plugin-lodash](https://github.com/idok/eslint-plugin-lodash).


Updates `@antv/g6` from 4.8.24 to 5.0.10
- [Release notes](https://github.com/antvis/g6/releases)
- [Changelog](https://github.com/antvis/G6/blob/v5/CHANGELOG.md)
- [Commits](https://github.com/antvis/g6/commits)

Updates `@types/node` from 20.14.13 to 22.0.0
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Updates `eslint-plugin-lodash` from 7.4.0 to 8.0.0
- [Changelog](https://github.com/idok/eslint-plugin-lodash/blob/master/CHANGELOG.md)
- [Commits](https://github.com/idok/eslint-plugin-lodash/commits/v8.0.0)

---
updated-dependencies:
- dependency-name: "@antv/g6"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: eslint-plugin-lodash
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update package.json

Signed-off-by: lijianan <574980606@qq.com>

* Update package.json

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: lijianan <574980606@qq.com>

magicdawn committed:
docs: Update README.md reproduce link (#50151)

Signed-off-by: Tao <magicdawn@qq.com>

zombieJ committed:
enhance: color picker click (#50148)

magicdawn committed:
Update README-zh_CN.md reproduce link (#50152)

Signed-off-by: Tao <magicdawn@qq.com>

wanpan11 committed:
fix: Tree filter node style lost (#49773)

github-actions[bot] committed:
chore: upgrade deps (#50180)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

sergioalmela committed:
fix: Improve es_ES.ts (#50179)

Add more natural language to Spanish translations

Signed-off-by: Sergio <sergioalmela@gmail.com>

coding-ice committed:
feat(spin): configProvider support indicator prop (#50183)

DDDDD12138 committed:
fix: update script for cross-platform compatibility (#50184)

Zyf665 committed:
fix(table):table grouping columns does not apply sorting (#50086)

* 增加函数循环收集mergedColumns中所有key

* 新增测试用例

* ci问题调整，收集列keys函数调整

afc163 committed:
docs: fix findDOMNode warnings (#50191)

* docs: fix findDOMNode warnings

* docs: fix findDOMNode warnings

* fix ts type

baicie committed:
chore:  change to export type (#50203)

gin-lsl committed:
docs: fix broken umi link (#50214)

vagusX committed:
docs: improve Segmented docs with SegmentedItemType (#50205)

* docs: improve Segmented docs with SegmentedItemType

* chore: update

* chore: update

* test: update snapshot

* docs: update

andrii-milishchuk committed:
fix: add missed translation in uk_UA (#50178)

* Update uk_UA.ts

Signed-off-by: Andrii Milishchuk <andrewmilishchuk@gmail.com>

* Update index.test.tsx.snap

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: Andrii Milishchuk <andrewmilishchuk@gmail.com>
Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: lijianan <574980606@qq.com>

vagusX committed:
fix: unit for line-height for ColorPicker (#50220)

* fix: unit for line-height for ColorPicker

* chore: update

coding-ice committed:
fix(color-picker): In disabled state, cursor is no-allowed (#50217)

vagusX committed:
test: set maxWorkers for test:visual-regression (#50219)

* test: set maxWorkers for test:visual-regression

* chore: print args

* chore: set max-workers to 4

* test: use move instead of copy

* chore: set max-workers to 2

vagusX committed:
docs: v5.20.0 (#50218)

* docs: v5.20.0

* docs: fix changelog lint

* chore: update

* chore: update

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: vagusX <vagusxl@gmail.com>

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: vagusX <vagusxl@gmail.com>

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: vagusX <vagusxl@gmail.com>

* chore: update

* docs: update

---------

Signed-off-by: vagusX <vagusxl@gmail.com>
Co-authored-by: afc163 <afc163@gmail.com>

FanetheDivine committed:
docs(Form): supplement the documentation of Form (#50223)

* docs: 对Form组件的文档做补充

* Apply suggestions from code review

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: Fane <141594291+FanetheDivine@users.noreply.github.com>

---------

Signed-off-by: Fane <141594291+FanetheDivine@users.noreply.github.com>
Co-authored-by: afc163 <afc163@gmail.com>

github-actions[bot] committed:
chore: upgrade deps (#50225)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @typescript-eslint/parser to v8 (#50228)

* chore(deps): update dependency @typescript-eslint/parser to v8

* Update package.json

Signed-off-by: afc163 <afc163@gmail.com>

* fix eslint errors

---------

Signed-off-by: afc163 <afc163@gmail.com>
Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: afc163 <afc163@gmail.com>

hzyhbk committed:
docs: update components/_util/index.zh-CN.md (#50235)

修改错别字

thinkasany committed:
fix(locale): spaces in default az_AZ language (#50238)

Meetacoo committed:
docs(ColorPicker): Update line-gradient.md (#50239)

Signed-off-by: Meetacoo <37367461+Meetacoo@users.noreply.github.com>

li-jia-nan committed:
chore: update componentCls key (#50244)

li-jia-nan committed:
feat: App support direction with CP (#50246)

* feat: App support direction with CP

* feat: App support direction with CP

vagusX committed:
test: fix test:visual-regression when moving imgs (#50253)

astfn committed:
docs: fix typo (#50256)

li-jia-nan committed:
chore: update componentCls key again (#50251)

* chore: update componentCls key again

* Update components/skeleton/style/index.ts

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: lijianan <574980606@qq.com>

Wxh16144 committed:
ci: antd 4.x eol (#50254)

li-jia-nan committed:
chore: update unit value (#50258)

Wxh16144 committed:
fix(Spin): progress bar mode in container not centered error (#50277)

crazyair committed:
doc: @ctrl/tinycolor (#50280)

guoyunhe committed:
feat(upload): showUploadList options showPreviewIcon, showDownloadIcon and showRemoveIcon support function (#50245)

zombieJ committed:
docs: fix limit demo (#50287)

coding-ice committed:
test(getScroll): When window is undef (#50266)

* test(getScroll): When window is undef

* test: document -> null

* test: mock

afc163 committed:
style(Skeleton): remove default children of Skeleton.Node (#50278)

* style(Skeleton): remove default children of Skeleton.Node

* style(Skeleton): remove default children of Skeleton.Node

* Update components/skeleton/style/index.ts

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: afc163 <afc163@gmail.com>

---------

Signed-off-by: afc163 <afc163@gmail.com>
Co-authored-by: lijianan <574980606@qq.com>

Asanio06 committed:
fix(Calendar): make Calendar locale overriding locale from ConfigProvider (#50236)

Co-authored-by: Lansana DIOMANDE <lansana.diomande@cartier.comm>

zombieJ committed:
fix: ColorPicker not support compact (#50291)

* fix: ColorPicker not support compact

* test: update snapshot

* test: update snapshot

afc163 committed:
fix: fast-color compile error by upgrade color-picker (#50293)

Signed-off-by: afc163 <afc163@gmail.com>

github-actions[bot] committed:
chore: upgrade deps (#50294)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

afc163 committed:
docs: add offcial pronunciation of antd (#50297)

afc163 committed:
chore: improve pr template (#50301)

* chore: update pr template

* chore: update pr template

* chore: update pr template

zombieJ committed:
fix: tour default zIndex (#50300)

vagusX committed:
chore: retry upload add delay (#50302)

zombieJ committed:
fix: form should update

zombieJ committed:
docs: fix site css sandbox open (#50305)

zombieJ committed:
docs: add watermark faq (#50307)

thinkasany committed:
chore(tree-select): improve code style by restProps (#50314)

zombieJ committed:
fix: tree switch bg height (#50313)

* fix: tree switch bg height

* fix: use before instead

li-jia-nan committed:
docs: update Version of variant (#50319)

github-actions[bot] committed:
chore: upgrade deps (#50321)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

zombieJ committed:
fix: Typography ellipsis (#50315)

* fix: get P

* test: add test case

li-jia-nan committed:
demo: site & demo update (#50322)

thinkasany committed:
type: ts definition improvement (#50324)

* type: improve type assert

* type: improve type assert

* chore: modify the path

thinkasany committed:
docs(modal): use Record<SemanticDOM> instead of classnames/styles type (#50330)

afc163 committed:
demo: improve typography copyable demo (#50336)

y-young committed:
docs(AutoComplete): add missing options in en-US document (#50340)

Signed-off-by: Googleplex <yyoung2001@gmail.com>

li-jia-nan committed:
feat: FloatButton support dynamic z-index (#50311)

* feat: FloatBtn support dynamic z-index

* feat: FloatBtn support dynamic z-index

* test: fix test case

* fix: fix

* test: add test case

* test: fix test case

* chore: clear

github-actions[bot] committed:
chore: upgrade deps (#50342)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

li-jia-nan committed:
type(Table): TypeScript improvement (#50351)

* type: Table TypeScript improvement

* type: Table TypeScript improvement

* type: Table TypeScript improvement

* fix: fix demo

* fix: fix demo

* fix: fix type

* fix: fix type

* fix: fix type

* fix: fix type

renovate[bot] committed:
chore(deps): update dependency cheerio to v1.0.0 (#50352)

* chore(deps): update dependency cheerio to v1.0.0

* Update check-site.ts

Signed-off-by: afc163 <afc163@gmail.com>

* Update package.json

Signed-off-by: afc163 <afc163@gmail.com>

* Update scripts/check-site.ts

Signed-off-by: afc163 <afc163@gmail.com>

---------

Signed-off-by: afc163 <afc163@gmail.com>
Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: afc163 <afc163@gmail.com>

afc163 committed:
test: speed up test workflow (#50346)

* test: faster test.yml

* test: upgrade to node 20

* fix depends

* fix needs

* fix needs

* fix needs

* fix needs

* test: update test.yml

* test: update test.yml

* test: update test.yml

* test: update test.yml

* test: update test.yml

* test: update test.yml

* test: update test.yml

* test: update test.yml

* test: update test.yml

* fix: lib es test

* fix: lib es test

* fix: lib es test

* skip dist test in react lagacy

* Update test.yml

Signed-off-by: afc163 <afc163@gmail.com>

---------

Signed-off-by: afc163 <afc163@gmail.com>

li-jia-nan committed:
fix(Table): style value should support CSS var (#50355)

* fix(Table): zIndex value should support CSS var

* fix: fix

afc163 committed:
chore: fix surge preview deploy not throw error (#50356)

Signed-off-by: afc163 <afc163@gmail.com>

li-jia-nan committed:
refactoring: rewrite renderCircle with React Component (#50358)

li-jia-nan committed:
fix(Table): set z-index unit to false (#50357)

afc163 committed:
chore: fix surge deploy by locking version to 0.23.1 (#50359)

afc163 committed:
chore: Update pr-open-notify.yml (#50361)

Signed-off-by: afc163 <afc163@gmail.com>

li-jia-nan committed:
chore: merge master into feature

afc163 committed:
docs(✨ ): release 5.20.1 (#50363)

* docs: release 5.20.1

* docs: release 5.20.1

* docs: release 5.20.1

* update changelog

* update changelog

* update changelog

* fix surge preview

* Update scripts/print-changelog.ts

Signed-off-by: afc163 <afc163@gmail.com>

---------

Signed-off-by: afc163 <afc163@gmail.com>

afc163 committed:
docs: fix github url typo (#50365)

li-jia-nan committed:
fix(site): fix lost indent of site (#50366)

* fix(site): lost indent of site

* fix: fix style

li-jia-nan committed:
fix(site): the img element of site should not be block (#50367)

Wxh16144 committed:
fix(popover): arrow. pointAtCenter misalignment error (#50260)

* fix(popover): 'arrow. pointAtCenter' misalignment error

* chore: add debug demo

* chore: update snaphots

* chore: update

* chore: Reverse components/popover/demo/basic.tsx

* chore: update snaphots

* chore: skip debug demo test

* chore: update demo

* chore: update snapshot

* chore: skip debug demo test

ref: https://github.com/ant-design/ant-design/pull/50260#discussion_r1708926635

afc163 committed:
docs: turn off dumi analyze in production mode (#50373)

afc163 committed:
chore: restore surge latest version (#50375)

afc163 committed:
fix: Select arrow disappearing when hovering it (#50382)

crazyair committed:
type: revert Table generic with any (#50372)

* fix: table generic default any

* feat: = AnyObject

arvinxx committed:
chore: Add comments to token interfaces (#50386)

afc163 committed:
test: enable codecov bundle analysis (#50384)

thinkasany committed:
docs(card): use card semantic-dom instead of table (#50390)

Co-authored-by: codefactor-io <support@codefactor.io>

huiliangShen committed:
fix: Upload.Dragger drag-and-drop upload directory issue (#50394)

thinkasany committed:
docs(card): remove unuesd gaps (#50393)

arvinxx committed:
chore: Add comments to Token related interfaces (#50392)

ohhoney1 committed:
docs(form): update link (#50401)

github-actions[bot] committed:
chore: upgrade deps (#50406)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

waywardmonkeys committed:
chore: reduce typo (#50404)

li-jia-nan committed:
fix: fix incorrect border-radius of trigger (#50408)

linxianxi committed:
chore: bump rc-table
